addons/mail/controllers/google_translate.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import requests
+import babel
+from odoo.http import request, route, Controller
+
+
+class GoogleTranslateController(Controller):
+    @route("/mail/message/translate", type="json", auth="user")
+    def translate(self, message_id):
+        user_lang = request.env.user.lang.split("_")[0]
+        translation = (
+            request.env["mail.message.translation"]
+            .sudo()
+            .search(
+                [("message_id", "=", message_id), ("target_lang", "=", user_lang)],
+                limit=1,
+            )
+        )
```

Comment:
This `sudo` is only acceptable if the current user has the rights to see the target message.

So the controller should first try to read the message without sudo and ensure it was allowed.

Reply:
Please add a test to cover the access right issue.